test/functionalTest/cases/0876_entity_dates/notify_entity_dates_with_attr.test

```diff
@@ -0,0 +1,951 @@
+                "type": "Number",
+                "value": 40
+            },
+            "dateCreated": {
+                "metadata": {},
+                "type": "DateTime",
+                "value": "REGEX(2\d\d\d\-.*)"
+            },
+            "dateModified": {
+                "metadata": {},
+                "type": "DateTime",
+                "value": "REGEX(2\d\d\d\-.*)"
+            },
+            "id": "E4",
+            "type": "T"
+        }
+    ],
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+
+
+05a. Create sub5 for E5 with attrs: dateModified,dateCreated,A
+==============================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+05b. Create entity E5 with attribute A and B
+============================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E5?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+05c. Update entity to trigger a second notification
+===================================================
+HTTP/1.1 204 No Content
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+05d. Dump and reset: see E5 with attributes A, dateCreated, dateModified (2 notifications)
+==========================================================================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 288
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A": {
+                "metadata": {},
+                "type": "Number",
+                "value": 5
+            },
+            "dateCreated": {
+                "metadata": {},
+                "type": "DateTime",
+                "value": "REGEX(2\d\d\d\-.*)"
+            },
+            "dateModified": {
+                "metadata": {},
+                "type": "DateTime",
+                "value": "REGEX(2\d\d\d\-.*)"
+            },
+            "id": "E5",
+            "type": "T"
+        }
+    ],
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Content-Length: 290
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A": {
+                "metadata": {},
+                "type": "Number",
+                "value": 500
+            },
+            "dateCreated": {
+                "metadata": {},
+                "type": "DateTime",
+                "value": "REGEX(2\d\d\d\-.*)"
```

Comment:
This REGEX partern should be used instead of REGEX(.*) for values meaning creation or modification dates. It ensures that the time starts with "2xxx-" so it can detect fails due to dates has not been initialized (the not-initialized value is 0, which corresponds to a date in 1970, so the pattern will detect the problem).

NTC